frontend/src/config/config.test.ts

```diff
@@ -0,0 +1,117 @@
+import jp from 'jsonpath';
+import { startCase } from 'lodash';
+import { Country, configMap, getRawLayers, getTranslation } from './index';
+import fg from 'fast-glob';
+import fs from 'fs';
+
+// Load all translation keys from the source code
+const translationKeyRegex = /[ {]t\('([^']+)'\)/g;
+const files = fg.sync('./src/**/*.{ts,tsx}');
+const translationKeys = new Set<string>();
+
+files.forEach(file => {
+  const content = fs.readFileSync(file, 'utf8');
+  let match;
+  // eslint-disable-next-line fp/no-mutation, no-cond-assign
+  while ((match = translationKeyRegex.exec(content)) !== null) {
+    translationKeys.add(match[1]);
+  }
+});
+
+describe('Config Map', () => {
+  it('should have necessary data for each country', () => {
+    Object.keys(configMap).forEach(country => {
+      const countryConfig = configMap[country as Country];
+      expect(countryConfig).toHaveProperty('defaultBoundariesFile');
+      expect(countryConfig).toHaveProperty('rawTables');
+      expect(countryConfig).toHaveProperty('rawReports');
+      expect(countryConfig).toHaveProperty('appConfig');
+      expect(countryConfig).toHaveProperty('rawLayers');
+    });
+  });
+
+  it('should have translations with correct depth', () => {
+    Object.keys(configMap).forEach(country => {
+      const config = configMap[country as Country];
+      const layers = getRawLayers(country as Country);
+      const translation = getTranslation(country as Country);
+
+      // All the elements need translation and should appear at least once in the translations:
+      /**
+       * - All the keys in config.categories
+       * - All the key
+       */
+      let itemsToTranslate: string[] = [];
+
+      // categories
+      const categoryKeys = jp
+        .query(config.appConfig.categories, '$.*')
+        .filter(
+          (value: any) =>
+            typeof value === 'string' ||
+            (typeof value === 'object' &&
+              value !== null &&
+              !Array.isArray(value)),
+        )
+        .map((value: any) =>
+          typeof value === 'object' ? Object.keys(value) : value,
+        )
+        .flat()
+        .map((value: any) => startCase(value));
+
+      // grouped layer titles
+      const categoryGroupTitles = jp.query(
+        config.appConfig.categories,
+        '$..group_title',
+      );
+
+      // layers titles
+      const layerTitles = jp.query(layers, '$..title');
+      const layerLegendTexts = jp.query(layers, '$..legend_text');
+      const chartLegendLabels = jp.query(
+        layers,
+        '$..chart_data.fields[*].label',
+      );
+      // const legendLabels = jp.query(layers, '$..legend[*].label');
+      // eslint-disable-next-line fp/no-mutation
+      itemsToTranslate = [
+        ...Array.from(translationKeys),
+        ...categoryKeys,
+        ...categoryGroupTitles,
+        ...layerTitles,
+        ...layerLegendTexts,
+        ...chartLegendLabels,
+      ];
```

Comment:
In particular I did not add legend labels because most of them are `20mm`, `23C`, ... and so on but let me know if you would like to include them and exclude simple labels with some patterns

Reply:
@ericboucher I'd rather add the legend labels as it's easy to miss content there that does require translation. Thanks!

Reply:
@wadhwamatic I added the labels but I feel like it adds quite a bit of noise. Let me know what you think.
<img width="854" alt="Screenshot 2024-07-12 at 11 31 01 PM" src="https://github.com/user-attachments/assets/e4019a62-5960-4449-8a82-87217f8c15fd">